plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/RepositoryReverseProxyFilter.java

```diff
@@ -0,0 +1,145 @@
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.docker.internal;
+
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.sonatype.nexus.common.app.BaseUrlHolder;
+import org.sonatype.nexus.common.app.BaseUrlManager;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.proxy.ProxyServlet;
+
+@Named
+@Singleton
+public class RepositoryReverseProxyFilter
+    implements Filter
+{
+
+  private final BaseUrlManager baseUrlManager;
+
+  private Map<Integer, String> portMappings = new ConcurrentHashMap<>();
+
+  private TransparentReverseProxyServlet reverseProxy = new TransparentReverseProxyServlet();
+
+  @Inject
+  public RepositoryReverseProxyFilter(final BaseUrlManager baseUrlManager) {this.baseUrlManager = baseUrlManager;}
+
+  @Override
+  public void init(final FilterConfig config) throws ServletException {
+    reverseProxy.init(new ServletConfig()
+    {
+      @Override
+      public String getServletName() {
+        return "reverse-proxy";
+      }
+
+      @Override
+      public ServletContext getServletContext() {
+        return config.getServletContext();
+      }
+
+      @Override
+      public String getInitParameter(final String name) {
+        return null;
+      }
+
+      @Override
+      public Enumeration<String> getInitParameterNames() {
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void destroy() {
+    // ignore
+  }
+
+  @Override
+  public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain)
+      throws IOException, ServletException
+  {
+    if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
+      HttpServletRequest httpRequest = (HttpServletRequest) request;
+      String repositoryName = portMappings.get(request.getLocalPort());
+      if (repositoryName != null) {
+        String uri = httpRequest.getRequestURI();
+        if (httpRequest.getQueryString() != null) {
+          uri = uri + "?" + httpRequest.getQueryString();
+        }
+        if (!uri.startsWith("/static")
+            && !uri.startsWith("/error.html")
+            && !uri.startsWith("/repository/" + repositoryName)) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/repository/" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)